java/code/src/com/redhat/rhn/domain/entitlement/VirtualizationEntitlement.java

```diff
@@ -63,7 +65,22 @@ public boolean isAllowedOnServer(Server server) {
      * {@inheritDoc}
      */
     public boolean isAllowedOnServer(Server server, ValueMap grains) {
-        return super.isAllowedOnServer(server) &&
-                grains.getOptionalAsString("virtual").orElse("physical").equals("physical");
+        String type = grains.getOptionalAsString("virtual").orElse("physical");
```

Comment:
Is it guaranteed that type is "lowercase" ? AFAIK this is the input requirement for `isVirtualGuest()`

Reply:
the type value is not guaranteed to be lowercase, but the `physical` and `xen` possible values we are checking here are lowercase in the Salt code. Nothing ensures this would never change in the future, but they tend to keep backward compatibility of that sort of things for two versions.